File: StructuralDesignPattern/ProxyPattern/java/Driver.java
package StructuralDesignPattern.ProxyPattern.java;

import java.util.List;

import StructuralDesignPattern.ProxyPattern.java.AbstractService.YoutubeService;
import StructuralDesignPattern.ProxyPattern.java.Model.Video;
import StructuralDesignPattern.ProxyPattern.java.Proxy.ProxyYoutubeService;

public class Driver {
    public static void main(String[] args){
        YoutubeService youtubeService = new ProxyYoutubeService();

        List<Video> popularVideos = youtubeService.getPopularVideos();
        for(Video video: popularVideos){
            System.out.println(video);
        }

        popularVideos = youtubeService.getPopularVideos();
        for(Video video: popularVideos){
            System.out.println(video);
        }

        Video video = youtubeService.getVideo("ABC");

        System.out.println(video);

        video = youtubeService.getVideo("ABC");

        System.out.println(video);
    }
}

File: StructuralDesignPattern/ProxyPattern/python/Driver.py
from Proxy.ProxyYoutubeService import ProxyYoutubeService


if __name__ == "__main__":

    proxyService = ProxyYoutubeService()

    popular_videos = proxyService.getPopularVideos()
    for video in popular_videos:
        print(video.getVideoDetails())

    print("\nFetching popular videos again:")
    popular_videos = proxyService.getPopularVideos()
    for video in popular_videos:
        print(video.getVideoDetails())


    print("\nFetching video with ID '1':")
    video = proxyService.getVideo("1")
    print(video.getVideoDetails())

    # Fetch the same video again (will return cached result)
    print("\nFetching video with ID '1' again:")
    video = proxyService.getVideo("1")
    print(video.getVideoDetails())